Library/Homebrew/download_strategy.rb

```diff
@@ -312,6 +329,10 @@ def _fetch
     curl @url, "-C", downloaded_size, "-o", temporary_path
   end
 
+  def does_redirect
```

Comment:
Not sure what I'm screwing up with your contribution Xu. I stripped all of my code out just to check and it's still throwing failure.

```ruby

  def fetch
    ohai "Downloading #{@url}"

    begin
      resolve_url
    end

  def resolve_url(url)
    m = Utils.popen_read("curl", "-I", url).match /^Location: (.+)$/
    m.nil? ? url : resolve_url(m.captures.first)
  end
```

Failure is: ` Error: wrong number of arguments (0 for 1) `.